RetrospectiveExtension.Frontend/components/feedbackCarousel.tsx

```diff
@@ -77,7 +79,7 @@ export default class FeedbackCarousel extends React.Component<IFeedbackCarouselP
             {mainCardCount >= 2 &&
               // @ts-ignore TS2786
               <Slider {...settings}>
-                {React.Children.map(this.renderFeedbackCarouselItems(columnProps), (child: React.ReactElement<FeedbackItem>) => {
+                {React.Children.map(this.renderFeedbackCarouselItems(columnProps), (child: React.ReactElement<typeof FeedbackItem>) => {
```

Comment:
That is a good question. I guess it is because `withAITracking` takes a class as input and returns a value instead of another class. Hence if we remove the `typeof`, the compiler would complain that a value is being used a class. In Typescript there are little difference between an actual class and an object that has the same fields/parameters as defined in a class. Expect in small ways such as typeof. The original function `React.ReactElement<T>` expected either a class or a string,
`interface ReactElement<P = any, T extends string | JSXElementConstructor<any> = string | JSXElementConstructor<any>>`  I am not sure why the function `withAITracking` in the app insights SDK does not return an object that implments `JSXElementConstructor` otherwise the `typeof` expression would have been needless. I can look deeper into this if needed.

Reply:
No need, thanks for your explanation!